### PR TITLE
Create a manually triggered workflow to release WooCommerce Beta Tester

### PR DESCRIPTION
**Pull request by samueljseay**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This is a MVP workflow for releasing beta tester. It is fairly manual. Once we have done the version bumping and code edits we are happy with to perform a release of beta tester this workflow can be initiated. My thought is that in future we can automate things like version bumping etc if we need to.

You just choose a version number for the release manually and start the workflow. It will create a release a tagged release on the monorepo that has a name and version unique to WooCommerce Beta Tester. It will never set the release as latest since we should reserve that just for releasing the core plugin.

### How to test the changes in this Pull Request:

1. Fork woocommerce and enable actions
2. Merge this PR into your fork
3. Go to Actions tab on your fork -> Find the action and run it, choosing an arbitrary version (e.g. `2.1.0` which is what the plugin version is right now)
4. It should succeed.
5. It should create a release under your forks "Releases" section which includes the beta tester plugin zip file as well.
6. The release should not be marked "latest"

For reference here is a passing action run on my fork: https://github.com/samueljseay/woocommerce/actions/runs/3898558244

and the created release: https://github.com/samueljseay/woocommerce/releases/tag/wc-beta-tester-2.4.0

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: .github/workflows/release-wc-beta-tester.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            - name: Setup WooCommerce Monorepo
+              uses: ./.github/actions/setup-woocommerce-monorepo
+
+            - name: Build WooCommerce Beta Tester Zip
+              working-directory: plugins/woocommerce-beta-tester
+              run: pnpm build:zip
+
+            - name: Create release
+              id: create_release
+              uses: samueljseay/action-gh-release@master
```
Comment (samueljseay) [inline .github/workflows/release-wc-beta-tester.yml:28]: I ended up forking the original action to add support for marking the release as "not latest". My plan is to upstream a PR for this fix to https://github.com/softprops/action-gh-release

File: .github/workflows/release-wc-beta-tester.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            - name: Setup WooCommerce Monorepo
+              uses: ./.github/actions/setup-woocommerce-monorepo
+
+            - name: Build WooCommerce Beta Tester Zip
+              working-directory: plugins/woocommerce-beta-tester
+              run: pnpm build:zip
+
+            - name: Create release
+              id: create_release
+              uses: woocommerce/action-gh-release@master
```
Comment (samueljseay) [inline .github/workflows/release-wc-beta-tester.yml:29]: I created a fork of the action, originally in my own account to test and add changes to support marking a release as "latest" or not. With this fork it now defaults to **not** marking the release as latest which you can see in my test run listed under test instructions in the PR description.

I have made an upstream PR to the original action so we could hopefully swap to their official version if/when it is merged: https://github.com/softprops/action-gh-release/pull/304

**Comment by roykho** [comment]
Hi @samueljseay - is this ready for review?

**Comment by samueljseay** [comment]
@roykho yup!

File: .github/workflows/release-wc-beta-tester.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+name: WooCommerce Beta Tester Release
+
+on:
+    workflow_dispatch:
+        inputs:
+            version:
+                description: 'The version number for the release'
+                required: true
+
```
Comment (roykho) [inline .github/workflows/release-wc-beta-tester.yml:10]: Would be a good idea to lock this workflow down initially with `permissions: {}` which gives zero permissions. Then unlock the minimum needed permissions in the jobs level which you have already done.

**Review by roykho — CHANGES_REQUESTED** [review]
Tested nicely. Good job! Just have a small suggested change.

File: .github/workflows/release-wc-beta-tester.yml (hunk 3 above)
Comment (samueljseay) [reply .github/workflows/release-wc-beta-tester.yml:10]: Good point! addressed in https://github.com/woocommerce/woocommerce/pull/36387/commits/ed342a78918bb65d0f53ed4c514b0e1185c8d57c

**Review by roykho — APPROVED** [review]
LGTM!

**Pull request merged**